#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Make glibc-2.3.5 compile with gcc-4.0 on hppa fixed by
#		   removing strong_alias.
# DP: Related bugs: 
# DP: Dpatch author: GOTO Masanori <gotom@debian.org> 
# DP: Patch author: GOTO Masanori <gotom@debian.org> 
# DP: Upstream status: Debian-Specific
# DP: Status Details: 
# DP: Date: 2005-08-03

PATCHLEVEL=0

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p$PATCHLEVEL < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p$PATCHLEVEL < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
2005-08-03  GOTO Masanori  <gotom@debian.org>

	* sysdeps/hppa/dl-machine.h: Remove strong_alias.

--- sysdeps/hppa/dl-machine.h.gotom	2005-08-04 23:43:40.000000000 +0900
+++ sysdeps/hppa/dl-machine.h	2005-08-04 23:47:35.000000000 +0900
@@ -538,8 +538,7 @@
   TRAMPOLINE_TEMPLATE (_dl_runtime_profile, profile_fixup);
 #else
 #define ELF_MACHINE_RUNTIME_TRAMPOLINE			\
-  TRAMPOLINE_TEMPLATE (_dl_runtime_resolve, fixup);	\
-  strong_alias (_dl_runtime_resolve, _dl_runtime_profile);
+  TRAMPOLINE_TEMPLATE (_dl_runtime_resolve, fixup);
 #endif
 
 /* ELF_RTYPE_CLASS_PLT iff TYPE describes relocation of a PLT entry, so
